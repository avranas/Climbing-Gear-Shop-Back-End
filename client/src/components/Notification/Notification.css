.notification-container {
  position: fixed;
  top: -2rem;
  width: 100%;
  display: flex;
  justify-content: center;
  animation: 2s ease-out 0s slideDownAndUp;
  animation-fill-mode: forwards;
  pointer-events: none;
}

@keyframes slideDownAndUp {
  10% {
    transform: translateY(150%);
  }
  100% {
    transform: translateY(150%);
  }
}

.notification-content {
  display: flex;
  align-items: center;
  pointer-events: auto;
  padding-left: 2rem;
  padding-right: 2rem;
  text-align: center;
  background-color: #c5ffbd;
  color: rgb(40, 61, 40);
  bottom: 5rem;
  border-radius: 5px;
  border: 1px solid rgba(12, 91, 0, 0.46);
  cursor: pointer;
}

.notification-content p {
  margin: 0.5rem;
  align-items: center;
}

.notification-icon {
  width: 1.5rem;
  height: 1.5rem;
  background-image: url('../../images/checkmark.png');
  background-size: cover;
  align-items: center;
}
