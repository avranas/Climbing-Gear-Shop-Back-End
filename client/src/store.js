import { configureStore } from "@reduxjs/toolkit";
import userReducer from './slices/userSlice';
import productListSlice from './slices/productListSlice';
import notificationSlice from './slices/notificationSlice';
import productSlice from "./slices/productSlice";
import cartSlice from "./slices/cartSlice";
import newestCartItemSlice from "./slices/newestCartItemSlice";

export default configureStore({
  reducer: {
    user: userReducer,
    productList: productListSlice,
    product: productSlice,
    notifications: notificationSlice,
    cart: cartSlice,
    newestCartItem: newestCartItemSlice
  },
});