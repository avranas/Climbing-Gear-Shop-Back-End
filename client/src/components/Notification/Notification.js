import { useDispatch } from "react-redux";
import "./Notification.css";
const Notification = (props) => {
  const dispatch = useDispatch();

  const clearNotification = () => {
    dispatch({
      type: "notifications/endNotification",
      payload: { id: props.id },
    });
  };

  return (
    <div className="notification-container">
      <div onClick={clearNotification} className="notification-content">
        <div className="notification-icon"></div>
        <p>{props.message}</p>
      </div>
    </div>
  );
};

export default Notification;
