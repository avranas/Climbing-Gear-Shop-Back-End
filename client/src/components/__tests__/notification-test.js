import { cleanup } from "@testing-library/react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import store from "../../store";
import Notification from "../Notification/Notification";

afterEach(() => {
  cleanup();
});

const notification = (
  <Provider store={store}>
    <BrowserRouter>
      <Notification />
    </BrowserRouter>
  </Provider>
);

test("Notification matches snapshot", async () => {
  const tree = renderer.create(notification).toJSON();
  expect(tree).toMatchSnapshot();
});