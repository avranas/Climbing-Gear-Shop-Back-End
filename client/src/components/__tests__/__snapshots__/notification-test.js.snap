// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification matches snapshot 1`] = `
<div
  className="notification-container"
>
  <div
    className="notification-content"
    onClick={[Function]}
  >
    <div
      className="notification-icon"
    />
    <p />
  </div>
</div>
`;
